src/shared/convert.js

```diff
@@ -1,35 +1,49 @@
 // @flow
 
-import type {ColorMatchType} from "./colors";
+import type {ColorMatchType} from './colors';
 import {colorArray, formats} from './colors';
 import type {TinyColor} from 'tinycolor2';
 import tinycolor from 'tinycolor2';
-import {ObjectMap} from "./generic";
+import {ObjectMap} from './generic';
+import diff from 'color-diff';
 
-function diffColors(from: string, to: string): number {
-  const materialRgb = tinycolor(from).toRgb();
-  const colorRgb = tinycolor(to).toRgb();
-  const diffColor = (key) => Math.abs(materialRgb[key] - colorRgb[key]);
-  return diffColor('r') + diffColor('g') + diffColor('b');
-}
+// `color-diff` only allows uppercase {R,G,B}
+// TODO: remove upcasing after submitting a PR for downcase
+type UpcaseRgbType = {
+  R: number,
+  G: number,
+  B: number
+};
+const hexToUpcaseRgb = function (hex: string): UpcaseRgbType {
+  const rgbColor = tinycolor(hex).toRgb();
+  return {R: rgbColor.r, G: rgbColor.g, B: rgbColor.b};
+};
 
-export function convert(color: string): ColorMatchType {
-  const colorMatch = colorArray.reduce((match: { diffVal: number, color: ColorMatchType }, colorItem: ColorMatchType) => {
-    const diffVal = diffColors(color, colorItem.value);
-    if (diffVal < match.diffVal) {
-      return {
-        diffVal,
-        color: colorItem
-      }
-    } else {
-      return match;
-    }
-  }, {
-    diffVal: Infinity,
-    color: ({}: any)
-  });
-  return colorMatch.color;
-}
+// Hoist upcase RGB objects so they're only created once (on first use)
+let colorMapRgb: Map<UpcaseRgbType, ColorMatchType>;
+let palette: Array<UpcaseRgbType>;
+
+export function convert(fromHex: string): ColorMatchType {
+  // Convert input color to upcase RGB object
+  const fromRgb = hexToUpcaseRgb(fromHex);
+
+  if (!colorMapRgb) {
+    // Create `Map` with pairs of `upcaseRgb: paletteColorObject`
+    colorMapRgb = new Map(
+      colorArray.map((colorObject) => [hexToUpcaseRgb(colorObject.value), colorObject])
+    );
```

Comment:
I'm left unclear from initial read why a `Map` is used here as opposed to a `reduce` + Object? Does the ordering of the `map` somehow increase the speed of the `diff.closest`?

If there's no performance gain and you're okay with it, I'd rather have a reduce as I find the `Map` implementation a little unclear personally 😅 (not due to anything of yours, just my silly mental blocks)

If we were to, would could redo this implementation as:
```javascript
const {obj, arr} = colorArray.reduce((prev, colorObject) => {
   const hexUpcaseRgbObjVal = hexToUpcaseRgb(colorObject.value);
   prev.obj[hexUpcaseRgbObjVal] = colorObject;
   prev.arr.push(hexUpcaseRgbObjVal);
   return prev;
}, {obj: {}, arr: []});

palette = arr;
colorMapRgb = obj;
```

And even knock out doing the `Array.from` later.

Reply:
@evelynhathaway just explained to me that 
> `hexUpcaseRgbObjVal` is an object so it cannot be a key which is why I made a Map 

Which makes perfect sense.

<details>
This is why we don't code review at night, folks
</details>